Functions/Public/New-GitHubGist.ps1

```diff
@@ -10,17 +10,56 @@ function New-GitHubGist {
       
     .Link
     https://trevorsullivan.net
+    http://dotps1.github.io
     https://developer.github.com/v3/gists
     #>
-    [CmdletBinding()]
-    param (
+    [CmdletBinding(DefaultParameterSetName = '__AllParameterSets')]
+    Param (
+        [Parameter(ParameterSetName = 'Files', Mandatory = $true, ValueFromPipeline = $true)]
+        [String[]]$Path,
+        [Parameter()]
+        [String]$Description,
+        [Parameter()]
+        [Switch] $Public
     )
-    
-    $ApiCall = @{
-        Body = '';
-        RestMethod = '';
-        Method = '';
+
+    DynamicParam {
+        if ($psISE -ne $null) {
+            $attribute = New-Object -TypeName System.Management.Automation.ParameterAttribute -Property @{
+                HelpMessage = 'Caputure the content of the current ISE tab.'
+                Mandatory = $true
+                ParameterSetName = 'IseScriptPane'
+            }
+            $collection = New-Object -TypeName System.Collections.ObjectModel.Collection[System.Attribute]
+            $collection.Add($attribute)
+            $parameter = New-Object -TypeName System.Management.Automation.RuntimeDefinedParameter('IseScriptPane', [Switch], $collection)
+            $dictionary = New-Object -TypeName System.Management.Automation.RuntimeDefinedParameterDictionary
+            $dictionary.Add('IseScriptPane', $parameter)
+            return $dictionary
+        }
     }
     
-    Invoke-GitHubApi @ApiCall;
+    Process {
+        [HashTable]$body = @{
+            description = $Description
+            public = $Public.IsPresent
+            files = @{}
+        }
+
+        if ($PSCmdlet.ParameterSetName -ne 'IseScriptPane') {
+            foreach ($item in $Path) {
+                $body.files.Add($(Split-Path -Path $item -Leaf), @{ content = ((Get-Content -Path $item -Raw).PSObject.BaseObject) })
```

Comment:
This is probably okay for v1, but we will need to do some validation on the file name and content. If there are any unsupported characters, or the content is too large, then we should "fail fast." :smile: